### DISCUSSION TITLE
Thermostat poll or get??? - Alarm.com integration

### DISCUSSION THREAD

**Discussion by iskoch01** [zwave-js/node-red-contrib-zwave-js#199] [Q&A] [answered]
Hi all, this package is incredibly helpful!

I have a question that maybe answered - can I put it in birds-eye view?

I have an older Simon Xt control panel - controlled via Alarm.com - controlling an Alarm.com thermostat - integrated to Node-Red with this excellent package and a Z-Stick 7. **The problem is with setpoint updates.**

Since I need current states when node-red launches, I got some help writing a query function. (Couldn't figure out cmd-factory. Help came from here: https://discourse.nodered.org/t/interrogating-z-wave-thermostat-using-node-red-contrib-zwave-js/56332/12)

When I changed the setpoints on my phone, the thermostat changed, but node-red didn't get updated. I could get the correct values only by interviewing the thermostat manually.

To work around the problem, I changed my function to pollValue instead of getValue.

Any ideas?

**Comment by marcus-j-davies** [comment]
Hi @iskoch01,

> To work around the problem, I changed my function to pollValue instead of getValue.

Using **pollValue** does make sense, as **getValue** returns last known, where as poll re-fetches from device.
using **pollValue** - is not normally needed, but on this occasion I can see why.

On the basis your problem is not getting updates in node red when you change it on you're phone:

This sounds like an association problem.
From Experience, thermos have odd association glitches, we usually have to patch them using configuration edits.

Can you confirm the model of the thermo, and also that it's lifeline is set to the controller (Node 1)

You can check the associations here:
Click the node in the UI, and open up Association Management - there should be a lifeline group and your zwave stick should be in there? if not add the controller => Node 1, Endpoint 0

**EDIT**
I have found a manual - that has an interesting section in value reporting - but want to to first check the obvious (and your model).
The manual I have found states the following for reporting values:

_Thermostat Setpoint Report : When either the heat or cool target **is changed locally**_

**Comment by iskoch01** [comment, marked as answer]
Thanks. I’ll check into it when I get back home to Nebraska this weekend. 

The thermostat is ADC-T2000.

When setting it up, I left the associations alone because the instructions I _was_ able to find was not enough.

I’m still puzzled why the thermostat sent updates to node red at first, but when alarm.com changed the set points, node red didn’t get it automatically. I’ll check into the associations.

**Reply by marcus-j-davies** [reply]
Hi @iskoch01 

found something of much interest (**bold**) - which may be the cause

**Config Parameter 35  -  Z-Wave Echo Association Reports**

_When enabled, the thermostat will report all changes to the thermostat mode, thermostat fan mode, and thermostat setpoint, **even when change was caused by commands from Z-wave.**
Size: 1 Byte, **Default Value: 0**_

if this is set to 0 - Disabled  (i.e the default) it wont report changes, that are made from a command (i.e those made by Alarm.Com)
its worth checking it.

 - Select the node
 - Collapse the Configuration CC
 - Check for this setting (Z-Wave Echo Association Reports)
 - set it to **enabled** 
 
 (missing associations was a guess - but the above is more likely)
 
 See here for params.
 https://devices.zwave-js.io/?jumpTo=0x0190:0x0001:0x0001:0.0

**Comment by iskoch01** [comment]
Interesting! I’ll check that, too! :)

**Comment by iskoch01** [comment]
The association was not set. Fixed. 
The echo reports now enabled. That seems to do the trick. Thank you!

Where can I go to learn more about association reports?

Thanks again.
